src/status_im/contexts/wallet/send/utils.cljs

```diff
@@ -67,10 +67,24 @@
              {}
              amounts))
 
+(defn add-zero-values-to-network-values
+  [network-values all-possible-chain-ids]
+  (reduce
+   (fn [acc chain-id]
+     (let [route-value (get network-values chain-id)]
+       (assoc acc chain-id (or route-value (money/bignumber "0")))))
+   {}
+   all-possible-chain-ids))
+
 (defn token-available-networks-for-suggested-routes
-  [{:keys [balances-per-chain disabled-chain-ids]}]
+  [{:keys [balances-per-chain disabled-chain-ids only-with-balance?]}]
   (let [disabled-set (set disabled-chain-ids)]
     (->> balances-per-chain
+         (filter (fn [[_ {:keys [raw-balance]}]]
+                   (or (not only-with-balance?)
+                       (and only-with-balance?
+                            (and (money/bignumber? raw-balance)
+                                 (money/greater-than raw-balance (money/bignumber "0")))))))
          (filter (fn [[_ {:keys [chain-id]}]]
                    (not (contains? disabled-set chain-id))))
```

Comment:
We can only use one `and` in the filter condition:
```clojure
(filter (fn [[_ {:keys [raw-balance]}]]
            (or (not only-with-balance?)
                (and only-with-balance?
                     (money/bignumber? raw-balance)
                     (money/greater-than raw-balance (money/bignumber "0"))))))
```

Have you consider extending the filter below instead of creating a new one? I think performing two filters is more expensive :thinking:

Reply:
Yeah, I've considered that option, thought that it was more "readable" even if it is a bit less performant, but I have no strong opinion on this tbh, I will change it to have only one filter :)

Reply:
We can create two functions to make it more readable, something like

```clojure
(filter (fn [my-data]
          (and (check-conditions-1 my-data)
               (check-conditions-2 my-data)))
```

If you take this approach, just make sure to provide good names :)